### PR TITLE
Editor Overscroll

### PR DESCRIPTION
**Pull request by yusufozgul**
Related Issue:
- #63

### CODE REVIEW

**Comment by lukepistrol** [comment]
Great work! 

One suggestion though:

- The current solution takes a fixed number of lines for the over scroll which might be more than the visible area of the text view. For example let's say the window has a height of 600pt and the over scroll line count is 70. For a font size of 13 pt this would result in a too large over scroll -> no content is visible once scrolled down all the way.

- Let's try to instead provide a double (0.0-1.0) which indicates in percent of the views visible height how large the over scroll is. This should also respect window size changes and recalculate the content inset based on that.

**Comment by yusufozgul** [comment]
Thank you for review :) 
 
Yes, you are right, I made changes, also I realized a bug while changing the content inset, I should have changed contentView's bottom inset not scrollView. Otherwise after updating once, the interface does not change

File: Sources/CodeEditTextView/CodeEditTextView.swift [hunk 1]
```diff
@@ -27,7 +27,7 @@ public struct CodeEditTextView: NSViewControllerRepresentable {
         font: Binding<NSFont>,
         tabWidth: Binding<Int>,
         lineHeight: Binding<Double>,
-        overScrollLineCount: Binding<Int> = .constant(0),
+        overScrollRatio: Binding<Double> = .constant(0.0),
```
Comment (lukepistrol) [inline Sources/CodeEditTextView/CodeEditTextView.swift:30]: please add `overScrollRatio` to the DocC comment annotation

**Review by lukepistrol — CHANGES_REQUESTED** [review]
Just a few documentation inconsistencies 

Let's also add the newly added symbols to the `Documentation.docc` catalog.
- `Sources/CodeEditTextView/Documentation.docc/CodeEditTextView.md`
- `Sources/CodeEditTextView/Documentation.docc/STTextViewController.md`

Also have a look at the failing SwiftLint rules

File: Sources/CodeEditTextView/CodeEditTextView.swift (hunk 1 above)
Comment (lukepistrol) [reply Sources/CodeEditTextView/CodeEditTextView.swift:30]: This still needs to be resolved

Comment (lukepistrol) [reply Sources/CodeEditTextView/CodeEditTextView.swift:30]: For clarification: I meant the doc comment above the init describing all the parameters.

File: Sources/CodeEditTextView/STTextViewController.swift [hunk 2]
```diff
@@ -43,7 +43,7 @@ public class STTextViewController: NSViewController, STTextViewDelegate, ThemeAt
     public var font: NSFont
 
     /// The overScrollLineCount to use for the textView over scroll
```
[the remark this answers is not in the document]
Comment (austincondiff) [reply Sources/CodeEditTextView/STTextViewController.swift]: @lukepistrol Why "ratio", isn't it representing a integer value and not a ratio?

**Comment by austincondiff** [comment]
This is my fault for not adding a description to the issue but I'd like to discuss how this works. I see now what @lukepistrol is suggesting. The first question is, does it make sense to allow that much flexibility?

As an example, Nova only has Small, Medium, and Large
image

VS Code only has a checkbox
image

My second question is, how are we displaying this in our Settings panel? 
If this is a percentage of editor viewport height, we should set this via a slider control
If this is a boolean, we should set this via a checkbox
And if we choose set values, we should handle it similar to Nova with a select input

**Comment by yusufozgul** [comment]
@austincondiff I actually thought Xcode like (none, small, medium, large). And even Nova the same. For ide, values will be just constant percentage.
Xcode percentages: 0.25, 0.5, 0.75

**Comment by lukepistrol** [comment]
@austincondiff I would either go with the way Xcode handles it (none, small, medium, large) or a slider with detents (0% - 100% in 10% steps).

A fixed number of lines becomes problematic when the window height becomes smaller than said amount. And a on-off toggle is not enough flexibility.

File: Sources/CodeEditTextView/STTextViewController.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (lukepistrol) [reply Sources/CodeEditTextView/STTextViewController.swift]: It's representing a Double (0.0-1.0) which is a percentage of the viewport height. So it is a quantitive relation between those values.

**Comment by matthijseikelenboom** [comment]
> Doesn't "amount" imply a number?

Yeah I agree. `editorOverscrollSetting` or `editorOverscrollConfig` comes to mind for me. But perhaps `editorOverscroll` is also fine.

**Comment by austincondiff** [comment]
I mean, "Editor Overscroll" is what the label says, so `editorOverscroll` makes sense to me

**Comment by yusufozgul** [comment]
It seems `editorOverscroll` nice for everyone

**Review by austincondiff — APPROVED** [review]

**Review by matthijseikelenboom — APPROVED** [review]

**Review by lukepistrol — APPROVED** [review]
LGTM 👍

**Pull request merged**